lib/bugsnag/middleware/session_data.rb

```diff
@@ -12,6 +12,7 @@ def call(report)
         else
           session[:events][:handled] += 1
         end
+        Bugsnag::SessionTracker.set_current_session session
```

Comment:
I dont understand why this is needed

Reply:
If multiple errors occur in a single session and are handled (i.e. not fatal) the session parameter should reflect that, showing the events incrementing over time.  Unless I'm fully misunderstanding how Ruby objects work, this should be necessary for that to happen.

Reply:
You shouldnt need to set it again. You are just mutating the object

Reply:
I didn't think that ruby objects worked like that, you're right, it's only PHP this is necessary.